packages/rollup-plugin/test/rollup-plugin.test.ts

```diff
@@ -33,46 +55,163 @@ async function buildAndMatchSnapshot(outputOptions: OutputOptions) {
 }
 
 describe('rollup-plugin', () => {
-  it('should build without preserveModules', async () => {
-    // Bundle all JS outputs together
-    await buildAndMatchSnapshot({
-      format: 'esm',
+  describe('options', () => {
+    it('extract generates .css bundle', async () => {
+      const cwd = path.dirname(
+        require.resolve('@fixtures/react-library-example/package.json'),
+      );
+      const output = await build({
+        cwd,
+        extract: { name: 'app.css', sourcemap: true },
+        rollup: {
+          input: { app: path.join(cwd, 'src/index.ts') },
+          external: ['clsx', 'react', 'react/jsx-runtime', 'react-dom'],
+          output: {
+            format: 'esm',
+            preserveModules: true, // not needed for output, just makes some assertions easier
+          },
+        },
+      });
+
+      // assert essential files were made
+      const bundleAsset = output.find(
+        (file) => file.type === 'asset' && file.fileName === 'app.css',
+      );
+      expect(bundleAsset).toBeTruthy();
+      const sourcemapAsset = output.find(
+        (file) => file.type === 'asset' && file.fileName === 'app.css.map',
+      );
+      expect(sourcemapAsset).toBeTruthy();
+
+      // assert .css imports were removed
+      const jsFiles = output.filter(
+        (file) => file.type === 'chunk' && file.fileName.endsWith('.js'),
+      ) as OutputChunk[];
+      for (const jsFile of jsFiles) {
+        expect(jsFile.code).not.toMatch(/^import .*\.css['"]/m);
+      }
+
+      // assert bundle CSS reflects order from @fixtures/react-library-example/index.ts
+      const map = JSON.parse(String((sourcemapAsset as OutputAsset).source));
+      expect(map.sources).toEqual([
+        'src/styles/reset.css.ts.vanilla.css',
+        'src/styles/vars.css.ts.vanilla.css',
+        'src/button/button.css.ts.vanilla.css',
+        'src/checkbox/checkbox.css.ts.vanilla.css',
+        'src/radio/radio.css.ts.vanilla.css',
+        'src/styles/utility.css.ts.vanilla.css', // this always should be last
+      ]);
+
+      // assert Vanilla CSS was stripped out
+      expect(
+        output.filter((file) => file.fileName.includes('.css.ts.vanilla')),
+      ).toHaveLength(0);
+
+      // snapshot output for everything else
+      expect(
+        output
+          .filter((chunkOrAsset) => !chunkOrAsset.fileName.endsWith('.map')) // remove .msps
+          .map((chunkOrAsset) => [
+            chunkOrAsset.fileName,
+            chunkOrAsset.type === 'asset'
+              ? chunkOrAsset.source
+              : chunkOrAsset.code,
+          ]),
+      ).toMatchSnapshot();
     });
   });
 
-  it('should build with preserveModules', async () => {
-    // Preserve JS modules
-    await buildAndMatchSnapshot({
-      format: 'esm',
-      preserveModules: true,
+  describe('Rollup settings', () => {
```

Comment:
These tests were not modified at all! Choose **Hide Whitespace** to see a better view